### DISCUSSION TITLE
Surface TX

### DISCUSSION THREAD

**Discussion by fraro** [ExpressLRS/ExpressLRS#1690] [Ideas]
Hi all,

in various forums the idea of building / converting a surface vehicle transmitter already came up, and is mostly brushed off with "better take a cheapo  30 bucks thing". But I prefer an open system over saving 2 bucks. And the latest ELRS PWM receivers are damn small and not really more expensive anymore. What would be the minimum requirements to implement, when not using OpenTX/Edge on the transmitter side? I'm fine with doing the configuration from outside (not via the handset). Thinking of interfacing an existing ELRS TX module with an RP2040 (for analog input from the pots and basic setup).
1. Implement CRSF protocol generation (any useful docs, arduino libs,....?)  
2. basic tx module control (enable)
3. ... ?
 
(maybe this can even be implemented on an STM8, which is used in FS-GT3C hanset)

Thanks a lot!

**Comment by mogplus88** [comment]
Very old thread and no response. 
I'm in the same boat. I have built a couple of transmitters with DIY encoders using Arduino Nano and FrSky hack module. The FrSky requires a PPM stream input. 
I would like to use an ELRS module, (and a Pi Pico) but I'm assuming it does not use a PPM stream as input. So what does it need? Is there a spec anywhere that describes what the input needs to look like? Is it a SPI protocol?
All advice gratefully received.
;-)

**Comment by sk8board** [comment]
RadioMaster has answered your request with the MT12.

https://www.radiomasterrc.com/collections/surface-radio
